niworkflows/viz/utils.py

```diff
@@ -66,6 +69,65 @@ def save_html(template, report_file_name, unique_string, **kwargs):
         handle.write(report_render)
 
 
+def svg_compress(image, compress='auto'):
+    ''' takes an image as created by nilearn.plotting and returns a blob svg.
+    Performs compression (can be disabled). A bit hacky. '''
+
+    # Compress the SVG file using SVGO
+    if (shutil.which("svgo") and compress == 'auto') or compress is True:
+
+        p = subprocess.Popen("svgo -i - -o - -q -p 3 --pretty --disable=cleanupNumericValues",
+                             stdin=subprocess.PIPE,
+                             stdout=subprocess.PIPE,
+                             shell=True)
+        image = p.communicate(input=image.encode('utf-8'))[0].decode('utf-8')
+        if p.wait() != 0:
+            raise Exception("svgo failed to run.")
+
+    # Convert all of the rasters inside the SVG file with 80% compressed WEBP
+    if (shutil.which("cwebp") and compress == 'auto') or compress == True:
+        new_lines = []
+        with StringIO(image) as fp:
+            for line in fp:
+                if "image/png" in line:
+                    tmp_lines = [line]
+                    while "/>" not in line:
+                        line = fp.readline()
+                        tmp_lines.append(line)
+                    content = ''.join(tmp_lines).replace('\n', '').replace(
+                        ',  ', ',')
+
+                    left = content.split('base64,')[0] + 'base64,'
+                    left = left.replace("image/png", "image/webp")
+                    right = content.split('base64,')[1]
+                    png_b64 = right.split('"')[0]
+                    right = '"' + '"'.join(right.split('"')[1:])
+
+                    bobj = base64.b64decode(png_b64)
+                    p = subprocess.Popen("cwebp -quiet -noalpha -q 80 -o - -- -",
+                                         stdin=subprocess.PIPE,
+                                         stdout=subprocess.PIPE,
+                                         shell=True)
+                    bimg = p.communicate(input=bobj)[0]
+                    if p.wait() != 0:
+                        raise Exception("cwebp failed to run.")
+                    new_lines.append(left + base64.b64encode(bimg).decode('utf-8') + right)
+                else:
+                    new_lines.append(line)
+        lines = new_lines
+    else:
+        with StringIO(image) as fp:
+            lines = fp.readlines()
```

Comment:
Is `StringIO` necessary here? Any reason this can't be `lines = image.splitlines()`?